torch/_inductor/codegen/cpp.py

```diff
@@ -1469,10 +1485,20 @@ def load(self, name: str, index: sympy.Expr):
         elif is_mask:
             line = f"flag_to_float_vec({loadbuf})"
         elif dtype in DTYPE_LOWP_FP:
-            line = f"at::vec::Vectorized<{DTYPE_TO_CPP[dtype]}>::loadu({loadbuf}, {self.tiling_factor})"
+            line = (
+                f"masked_load({loadbuf}, {load_mask})"
+                if load_mask
+                else f"at::vec::Vectorized<{DTYPE_TO_CPP[dtype]}>::loadu({loadbuf}, {self.tiling_factor})"
+            )
         else:
-            line = f"at::vec::Vectorized<float>::loadu({loadbuf})"
+            line = (
+                f"masked_load({loadbuf}, {load_mask})"
+                if load_mask
+                else f"at::vec::Vectorized<float>::loadu({loadbuf})"
+            )
+
         if non_contiguous:
+            # TODO: support masked_load for non_contiguous path?
```

Comment:
Are you going to handle this in this PR or a separate one? Maybe add an assertion here first to make sure we are not hitting the unexpected case?

Reply:
we can handle it at next step.